src/coreclr/scripts/antigen_setup.py

```diff
@@ -0,0 +1,158 @@
+    coreroot_dst_directory = path.join(correlation_payload_directory, "CoreRoot")
+    antigen_dst_directory = path.join(correlation_payload_directory, "exploratory")
+
+    helix_source_prefix = "official"
+    creator = ""
+    ci = True
+    if is_windows:
+        helix_queue = "Windows.10.Arm64" if arch_name == "arm64" else "Windows.10.Amd64.X86"
+    else:
+        if arch_name == "arm":
+            helix_queue = "(Ubuntu.1804.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7-bfcd90a-20200121150440"
+        elif arch_name == "arm64":
+            helix_queue = "(Ubuntu.1804.Arm64)Ubuntu.1804.ArmArch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652"
+        else:
+            helix_queue = "Ubuntu.1804.Amd64"
+
+    # create exploratory directory
+    print('Copying {} -> {}'.format(scripts_src_directory, coreroot_dst_directory))
+    copy_directory(scripts_src_directory, coreroot_dst_directory, match_func=lambda path: any(path.endswith(extension) for extension in [".py"]))
+
+    if is_windows:
+        acceptable_copy = lambda path: any(path.endswith(extension) for extension in [".py", ".dll", ".exe", ".json"])
+    else:
+        # Need to accept files without any extension, which is how executable file's names look.
+        acceptable_copy = lambda path: (os.path.basename(path).find(".") == -1) or any(path.endswith(extension) for extension in [".py", ".dll", ".so", ".json", ".a"])
+
+    # copy CORE_ROOT
+    print('Copying {} -> {}'.format(coreclr_args.core_root_directory, coreroot_dst_directory))
+    copy_directory(coreclr_args.core_root_directory, coreroot_dst_directory, match_func=acceptable_copy)
+
+    try:
+        with TempDir() as tool_code_directory:
+            # clone the tool
+            run_command(
+                ["git", "clone", "--quiet", "--depth", "1", "https://github.com/kunalspathak/Antigen.git", tool_code_directory])
+
+            antigen_bin_directory = path.join(tool_code_directory, "bin", "Release", "net5.0")
+
+            # build the tool
+            with ChangeDir(tool_code_directory):
+                dotnet_cmd = os.path.join(source_directory, "dotnet.cmd")
+                if not is_windows:
+                    dotnet_cmd = os.path.join(source_directory, "dotnet.sh")
+                run_command([dotnet_cmd, "publish", "-c", "Release", "--self-contained", "-r", run_configuration, "-o", antigen_bin_directory], _exit_on_fail=True)
+
+            if not os.path.exists(path.join(antigen_bin_directory, "Antigen.dll")):
+                raise FileNotFoundError("Antigen.dll not present at {}".format(antigen_bin_directory))
+
+            # copy antigen tool
+            print('Copying {} -> {}'.format(antigen_bin_directory, antigen_dst_directory))
+            copy_directory(antigen_bin_directory, antigen_dst_directory, match_func=acceptable_copy)
```

Comment:
I wonder if this should all be encapsulated in a script in the antigen source directory, so this script doesn't need to know how to build the tool. Something like `<tool_code_directory>build.cmd <dotnet_tool/script_to_use> <antigen_dst_directory>`.

Reply:
Sure, I will fix it in antigen and make related change here as a follow-up PR.